quisp/modules/QRSA/HardwareMonitor/HardwareMonitor.h

```diff
@@ -75,12 +85,15 @@ class HardwareMonitor : public IHardwareMonitor {
 
   virtual std::unique_ptr<NeighborInfo> createNeighborInfo(const cModule &thisNode);
   virtual std::unique_ptr<NeighborInfo> getNeighbor(cModule *qnic_pointer);
+  virtual cModule *getQNodeWithAddress(int address);
   virtual InterfaceInfo getQnicInterfaceByQnicAddr(int qnic_index, QNIC_type qnic_type);
   virtual void sendLinkTomographyRuleSet(int my_address, int partner_address, QNIC_type qnic_type, int qnic_index, unsigned long rule_id);
   virtual QNIC search_QNIC_from_Neighbor_QNode_address(int neighbor_address);
   virtual Matrix4cd reconstruct_Density_Matrix(int qnic_id);
+  virtual Matrix4cd extended_reconstruct_Density_Matrix(int qnic_id, int partner);
```

Comment:
method names should be started with verbs and snake_case. in this case, how about `reconstruct_extended_density_matrix` ?

Reply:
Yeah, let me use that name.